pkg/regexp/regexp.go

```diff
@@ -0,0 +1,210 @@
+package regexp
+
+import (
+	"io"
+	"regexp"
+	"sync"
+)
+
+// Regexp is a wrapper struct used for wrapping MustCompile regex expressions
+// used init. Using this stucture should help speed up startup time of apps
+// that want to initialize their regex at init time.
+type Regexp struct {
+	once   sync.Once
+	regexp *regexp.Regexp
+	val    string
+}
+
+func New(val string) Regexp {
```

Comment:
Name this `Delayed`, or something like that, to make it very clear at the call site that something unusual is going on and that this is not the standard library `regexp` package?

Alternatively, that could be expressed in the subpackage name.

Reply:
Sure, that works great.

Reply:
I just switched to Delayed()
I like regexp, for the conflict, to basically convince users to use it instead.  It should work fine even if the users compile the regex within a function and not a global.

Reply:
The only conflict I could see would be for callers of `regexp.QuoteMeta` … but that might also be something to nudge users away from.